pkg/messaging/v1/subscriptions_client.go

```diff
@@ -94,6 +102,35 @@ func (c *subscriptionsClient) UpdateSubscription(ctx context.Context, subscripti
 	return knerrors.GetError(err)
 }
 
+func (c *subscriptionsClient) UpdateSubscriptionWithRetry(ctx context.Context, name string, updateFunc SubscriptionUpdateFunc, nrRetries int) error {
+	return updateSubscriptionWithRetry(ctx, c, name, updateFunc, nrRetries)
+}
+
+func updateSubscriptionWithRetry(ctx context.Context, c KnSubscriptionsClient, name string, updateFunc SubscriptionUpdateFunc, nrRetries int) error {
+	b := retry.DefaultRetry
+	b.Steps = nrRetries
+	err := retry.RetryOnConflict(b, func() error {
+		return updateSubscription(ctx, c, name, updateFunc)
+	})
```

Comment:
```suggestion
	err := retry.RetryOnConflict(b, updateSubscription)
	})
```

Same as above. I haven't checked every occurrence of this pattern, so you might want to update the other locations, too.

Reply:
Again, please ignore my comment if the signature do no match. Sorry, should have seen that.